### DISCUSSION TITLE
Create Database connection lazily

### DISCUSSION THREAD

**Discussion by diptanu** [SeaQL/sea-orm#1485] [General]
Is it possible to create the connection lazily whenever the first request is made and thereafter cache the connection? And if it fails, I would like to retry again the next time a query has to be made. 

The reason is - I would like the service to start up even while the database is not responding. I can do this on my own by having an Option<DatabaseConnection> and then having a thread create the connection until it succeeds with a jittered delay if there are failures, but wondering if the framework makes this easier or if there are common patterns for this.

**Comment by tyt2y3** [comment]
There is a `min_connections` option and you can set it to `0`. This *should* do the trick.

**Reply by diptanu** [reply]
Interesting, I will try and also read the code. Would be nice if this was documented since I would imagine this would be something other users would need to consider as well.
